src/main/java/com/redhat/devtools/intellij/common/actions/AbstractActionVisibility.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.common.actions;
+
+import java.util.Arrays;
+
+abstract public class AbstractActionVisibility implements ActionVisibility {
+
+    @Override
+    public boolean isVisible(Object[] selected) {
+        if (selected == null
+                || selected.length == 0) {
+            return true;
+        }
+        return Arrays.stream(selected)
+                .map(this::map)
+                .anyMatch(this::isMatchingClasses);
+    }
+
+    protected abstract boolean isMatchingClasses(Object mapped);
```

Comment:
Because this class is abstract, it may not be based on classes so method should be better named `isMatching` ?

Reply:
dito